XCUITests/FxScreenGraph.swift

```diff
@@ -670,6 +674,35 @@ func createScreenGraph(for test: XCTestCase, with app: XCUIApplication) -> MMScr
             app.switches["ASPocketStoriesVisible"].tap()
         }
 
+
+        screenState.gesture(forAction: Action.SelectTopSitesRows_1) {userState in
+            //screenState.onEnterWaitFor(element: app.tables.cells["TopSitesRows"])
+            app.tables.cells["TopSitesRows"].tap()
+            app.staticTexts["1"].tap()
+            app.navigationBars.element(boundBy: 0).buttons.element(boundBy: 0).tap()
+        }
+
+        screenState.gesture(forAction: Action.SelectTopSitesRows_2) {userState in
+            //screenState.onEnterWaitFor(element: app.tables.cells["TopSitesRows"])
+            app.tables.cells["TopSitesRows"].tap()
+            app.staticTexts["2"].tap()
+            app.navigationBars.element(boundBy: 0).buttons.element(boundBy: 0).tap()
+        }
+
+        screenState.gesture(forAction: Action.SelectTopSitesRows_3) {userState in
+            //screenState.onEnterWaitFor(element: app.tables.cells["TopSitesRows"])
+            app.tables.cells["TopSitesRows"].tap()
+            app.staticTexts["3"].tap()
+            app.navigationBars.element(boundBy: 0).buttons.element(boundBy: 0).tap()
+        }
+
+        screenState.gesture(forAction: Action.SelectTopSitesRows_4) {userState in
```

Comment:
I think we can do this in a better way so that there is not code duplication. We are doing the same 4 times, just changing one small thing. We could create a userState and modify the number of rows selected before each `Action.SelectTopSitesRows`. For example `userState.numTopSites` in the test it would be something like:
```
userState.numTopSitesRows = 1
navigator.performAction(Action.SelectTopSitesRows)
```

And in the screengraph:
```
screenState.gesture(forAction: Action.SelectTopSitesRows) { userState in 
           app.tables.cells["TopSitesRows"].tap() 
           select(rows: userState.numTopSitesRows)
            app.navigationBars.element(boundBy: 0).buttons.element(boundBy: 0).tap()
        }
```
Where  `select(rows: userState.numTopSitesRows)`should be something like:
```
func select(rows: Int) {
        app.staticTexts[String(rows)].tap()
    }
```
This way we have to create only one Action.SelectTopSitesRows.

Hope this makes sense, please let me know otherwise :)

Reply:
There is an example about a similar implementation in the [Authentication tests](https://github.com/mozilla-mobile/firefox-ios/blob/master/XCUITests/AuthenticationTest.swift#L27)